QRHunter/app/src/main/java/com/example/qrhunter/utils/PlayerUtil.java

```diff
@@ -52,64 +59,112 @@ public static Map<String, Object> convertPlayerToHashmap(Player player) {
     /**
      * Calculate the total score given the list of qr codes
      *
-     * @param qrCodes The qr codes where their score will be calculated
+     * @param user The reference to the user to obtain scores from
      * @return The sum of the scores
      */
-    public double calculateTotalScore(ArrayList<QRCode> qrCodes) {
-        double currentTotalScore = 0;
-        for (int i = 0; i < qrCodes.size(); i++) {
-            QRCode qrcode = qrCodes.get(i);
-            currentTotalScore = currentTotalScore + qrcode.getScore();
-        }
-
-        return currentTotalScore;
+    public static double calculateTotalScore(String user) {
+        final double[] currentTotalScore = {0};
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+        db.collection("players")
+                .whereEqualTo("username", user)
+                .get()
+                .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
```

Comment:
I could be wrong but I don't think we are supposed to access firestore from the util, maybe pass a document snapshot or use the repository class?

Reply:
i just saw max's comment. u can just ignore this

Reply:
Sounds good I'll make that change ty